internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,257 @@
+		}
+		if isDeployed {
+			deployedEnvs = append(deployedEnvs, env.Name)
+		}
+	}
+
+	return deployedEnvs, nil
+}
+
+func (o *localRunOpts) isEnvironmentDeployed(envName string) (bool, error) {
+	var checker versionCompatibilityChecker
+
+	checker, err := o.envVersionChecker(envName)
+	if err != nil {
+		return false, err
+	}
+
+	currVersion, err := checker.Version()
+	if err != nil {
+		return false, fmt.Errorf("get environment %q version: %w", envName, err)
+	}
+	if currVersion == version.EnvTemplateBootstrap {
+		return false, nil
+	}
+	return true, nil
+}
+func (o *localRunOpts) validateEnvName() error {
+	isDeployed, err := o.isEnvironmentDeployed(o.envName)
+	if err != nil {
+		return err
+	}
+	if !isDeployed {
+		return fmt.Errorf(`cannot use an environment which is not deployed Please run "copilot env deploy, --name %s" to deploy the environment first`, o.envName)
+	}
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskWorkloadName() error {
+	if o.wkldName != "" {
+		return o.validateWkldName()
+	}
+	services, err := o.deployStore.ListDeployedServices(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("Get services: %w", err)
+	}
+
+	jobs, err := o.deployStore.ListDeployedJobs(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("Get jobs: %w", err)
+	}
+
+	workloads := append(services, jobs...)
+	if len(workloads) == 0 {
+		return fmt.Errorf("no workloads found in this environment %s", o.envName)
+	}
+	if len(workloads) == 1 {
+		log.Infof("Only one workload found in this environment, defaulting to: %s\n", color.HighlightUserInput(workloads[0]))
+		o.wkldName = workloads[0]
+	} else {
+		selectedWorloadName, err := o.prompt.SelectOne("Select a workload that you want to run locally", "", workloads, prompt.WithFinalMessage("workload name"))
+		if err != nil {
+			return fmt.Errorf("select Workload: %w", err)
+		}
+		o.wkldName = selectedWorloadName
+	}
+	localWorkloads, err := o.ws.ListWorkloads()
+	if err != nil {
+		return fmt.Errorf("list workloads in the workspace : %w", err)
+	}
+	if !contains(o.wkldName, localWorkloads) {
+		return fmt.Errorf("selected service %q does not exist in the workspace", o.wkldName)
+	}
+	return nil
+}
+
+func (o *localRunOpts) validateWkldName() error {
+	names, err := o.ws.ListWorkloads()
+	if err != nil {
+		return fmt.Errorf("list workloads in the workspace %s : %w", o.wkldName, err)
+	}
+	if !contains(o.wkldName, names) {
+		return fmt.Errorf("service %q does not exist in the workspace", o.wkldName)
+	}
+
+	if _, err := o.store.GetWorkload(o.appName, o.wkldName); err != nil {
+		return fmt.Errorf("Workload %q does not exist in smm", o.wkldName)
+	}
+	return nil
+}
+
+// BuildLocalRunCmd builds the command for running a workload locally
+func BuildLocalRunCmd() *cobra.Command {
+	vars := localRunVars{}
+	cmd := &cobra.Command{
+		Use:    "local run",
+		Short:  "Run the workload locally",
+		Long:   "Run the workload locally for debugging in a simulated AWS environment",
```

Comment:
```suggestion
		Long:   "Run the workload locally for debugging in a simulated AWS environment",
```
I am not sure about the "simulated". It sounds like we are simulating the VPC, db, etc. but we are not right?